src/main/java/com/sap/sgs/phosphor/fosstars/model/score/WeightedCompositeScore.java

```diff
@@ -159,20 +160,22 @@ public int hashCode() {
    * @param values The set of values.
    * @return A value of the specified score.
    */
-  private static ScoreValue calculate(Score score, ValueHashSet values) {
+  private static ScoreValue calculateIfNecessary(Score score, ValueHashSet values) {
     Optional<Value> something = values.of(score);
 
-    // if the set of values already contains a value for the specified score, then just return it
-    if (something.isPresent()) {
-      Value value = something.get();
-      if (value instanceof ScoreValue == false) {
-        throw new IllegalArgumentException("Hey! I expected a ScoreValue for a score!");
-      }
-      return  (ScoreValue) value;
+    // if the set of values doesn't contains a value for the specified score, then calculate it
+    Value value = something.orElseGet(() -> UnknownValue.of(score));
+    if (value.isUnknown()) {
+      return score.calculate(values);
+    }
+
+    // if the set of values contains a value for the specified score, then return it
+    if (value instanceof ScoreValue) {
+      return (ScoreValue) value;
     }
 
-    // otherwise, try to calculate a value for the score
-    return score.calculate(values);
+    throw new IllegalArgumentException(String.format(
```

Comment:
How about adding @throws in the javadoc for this exception?

Reply:
Yeah, why not.